cmd/kube-proxy/app/server.go

```diff
@@ -458,6 +462,20 @@ func serveMetrics(bindAddress string, proxyMode kubeproxyconfig.ProxyMode, enabl
 	}
 
 	configz.InstallHandler(proxyMux)
+	nodeIPs, err := proxyutil.FilterInterfaceAddrsByCIDRStrings(proxyutil.RealNetwork{}, cidrStrings)
+	if err != nil {
+		logger.Error(err, "failed to get node IPs for metrics server")
+	}
+	if len(nodeIPs) == 0 {
+		logger.Info("failed to get any node ip matching metricsBindAddresses", "metricsBindAddresses", cidrStrings)
+	}
+	var addrs []string
+	for _, nodeIP := range nodeIPs {
+		if nodeIP.IsLinkLocalUnicast() || nodeIP.IsLinkLocalMulticast() {
+			continue
+		}
```

Comment:
why?

Reply:
I'm getting this when binding local unicast IP addresses.

```listen tcp [fe80::42:acff:fe12:5]:10256: bind: invalid argument```

Reply:
ah, right, if you wanted to bind ipv6 link-local you'd need to specify which interface to bind it on, since the same link-local subnet is used by all interfaces